packages/shim-deno/src/deno/internal/Conn.ts

```diff
@@ -40,6 +41,38 @@ export class Conn extends FsFile implements Deno.Conn {
   unref(): void {
     this.#socket.unref();
   }
+
+  async read(p: Uint8Array): Promise<number | null> {
+    let wait = false;
+    while (true) {
+      if (wait) {
+        try {
+          await once(this.#socket, "readable", {
+            signal: AbortSignal.timeout(5),
```

Comment:
Do we need the abort signal? 5ms is very short.

Reply:
I am not sure if it's needed, but what curious would happen if that `readable` event was emitted before the `once` is called on it. Data loss?